fuel-core/src/database.rs

```diff
@@ -301,10 +301,11 @@ impl InterpreterStorage for Database {
 
     fn timestamp(&self, height: u32) -> Result<Word, Self::DataError> {
         let id = self.block_hash(height)?;
-        let block = self.storage::<FuelBlocks>().get(&id)?.unwrap_or_default();
+        // let block = self.storage::<FuelBlocks>().get(&id)?.unwrap_or_default();
```

Comment:
Not sure what to do here?

Reply:
Hmm yeah there is some edge cases that need to be handled here:

- If we request `block_hash(0)`, it will just return a default Bytes32::zeroed(). Since we temporarily store the currently executing block in the db at this same hash while the txs are being mutated, this will always return the current time. This is likely incorrect, as `timestamp(0)` should probably reflect the time of the genesis block (which would have to come from the chain_config instead of the DB).
- If we request the time of some block in the future that doesn't exist yet, (e.g.`timestamp(10)` when current height is 1), this will also return a zeroed block id which would refer to the currently executing block's timestamp. However, according to the specs, requesting the time of a block in the future should cause a [panic]

(https://github.com/FuelLabs/fuel-specs/blob/master/specs/vm/instruction_set.md#time-timstamp-at-height):
  > Panic if:
  >  - $rB is greater than the current block height.
  - Would need to check if this path is also triggered when the time is requested for the current block height, depending on whether or not the height index is updated to point current_height -> Bytes32::zeroed() during block execution.
- If we call these methods outside of the VM / block execution context, then both of these cases would return a default timestamp rather than the currently executing block, since there will be no block with a hash of Bytes32::zeroed() in the DB. Moving this to unwrap() would cause a panic, when we should either use genesis block time if height is 0, or a well-formed error if height > current height. The only case where a panic would make sense is when 0 < height <= current_block_height, since the block should definitely exist.

My suggestion would be to leave the behavior as it was, since addressing these edge cases seems orthogonal to updating the block header definition. We can file a bug to investigate and fix any behavioral discrepancies in a future PR.

Reply:
Ok I will file an issue for this